core/primitives/tests/crate-limit-test.rs

```diff
@@ -31,11 +22,40 @@ fn test_crate_count() {
         let crate_name = &cap[1];
         let crate_version = &cap[2];
         let crate_str = format!("{}-{}", crate_name, crate_version);
-        println!("{}", crate_str);
         unique_crates.insert(crate_str);
     }
+    println!("{:#?}", unique_crates);
     let crate_count = unique_crates.len();
     println!("Unique crate count: {}", crate_count);
 
-    assert!(crate_count < THRESHOLD, "Crate count is too high: {} > {}", crate_count, THRESHOLD);
+    assert!(crate_count < threshold, "Crate count is too high: {} > {}", crate_count, threshold);
+}
+
+#[test]
+fn test_crate_count() {
+    // Run `cargo tree -p near-primitives --edges=normal` and capture the output
+    let output = Command::new("cargo")
```

Comment:
Preexisting, but the right thing to do here would be to use the `CARGO` environment variable, rather than `cargo`.

Reply:
fixed with using compile time env! macro. I think it's pretty much resolves the situation as tests are run on same platform as compiled :)

Reply:
I'm suspicious of using `env!` here: in principle nothing requires the tests to run on the same machine that they were compiled on. But I guess if anybody is doing this, we'll find out soon enough...

Reply:
Well, potentially, but highly unluckily. Though, I don't mind to use just std::env::var with fallback to "cargo"